### PR TITLE
Curate new FAQ sections and add more answers

### PR DESCRIPTION
**Pull request by carolstran**
💓  **This PR is part of [Google Season of Docs](https://foundation.graphql.org/news/2020/09/21/google-season-of-docs-2020-participant-carolyn-stransky/)** 💓 

This PR breaks the existing questions into 5 sections:
- Getting Started
- General
- Best Practices
- Specification
- Frontend

👩🏼‍🎨  ⚠️  _The final design will be implemented in a separate PR to this branch, so **please focus on reviewing the content**._ 

Here are the new questions and answers I've added, sorted by section:

_**General**_
- [x] Does GraphQL use HTTP?
- [x] How does GraphQL affect my product’s performance?
- [x] Does GraphQL replace ORMs?
- [x] Who is behind GraphQL?

_**Best Practices**_
- [x] Is GraphQL scalable?
- [x] Does GraphQL support offline usage?
- [x] What are the security concerns with GraphQL?
- [x] How can I set up authorization with GraphQL?
- [x] How does authentication work with GraphQL?
- [x] Is GraphQL the right fit for designing a microservice architecture?
- [x] How does versioning work in GraphQL?
- [x] How can I document my GraphQL API?

_**Specification**_
- [x] What is the best way to follow specification releases?
- [x] Where is the documentation for subscriptions?

_**Frontend**_
- [x] Does GraphQL replace Redux or other state management libraries?

### CODE REVIEW

File: src/content/faq/General.md
Comment (benjie) [inline src/content/faq/General.md:151]: I think it's important to note that payload size of existing queries does not grow when you add fields/types to the schema.

```suggestion
There’s nothing that will prevent a GraphQL service from being versioned like any other REST API. That said, GraphQL avoids versioning by design. Instead, GraphQL provides the tools to continually build and evolve your schema. For example, GraphQL only returns the data that’s explicitly requested. This means that you can add new features (and all the associated types and fields) without creating a breaking change or bloating results for existing queries.
```

**Review by benjie — APPROVED** [review]
Looks good to me; couple suggestions but I approve :+1:

File: src/content/faq/General.md
Comment (mikeesto) [inline src/content/faq/General.md:113]:
```suggestion
GraphQL comes with some [built-in performance boosts](#how-does-graphql-affect-my-products-performance) that can help. But once you push it to production, you're responsible for scaling it across instances and monitoring performance.
```

File: src/content/faq/General.md
Comment (carolstran) [reply src/content/faq/General.md:97]: Ah that was my misinterpretation then! Thanks for clearing that up. I'll update 😁

File: src/content/faq/General.md
Comment (carolstran) [reply src/content/faq/General.md:113]: Not sure why, but it only seems to work with `product-s` and not `products` 🤷🏼‍♀️

**Review by Urigo — APPROVED** [review]
amazing work as usual.
I'm very excited you highlighted the Authentication and Authorization parts!

**Comment by carolstran** [comment]
Marking this as `On hold` until #985 is merged, then I'll do a big merge with all of the updates. If anyone wants to review that PR, I'd appreciate it 🙏🏼

File: src/components/FAQLayout/index.tsx
Comment (IvanGoncharov) [inline src/components/FAQLayout/index.tsx:22]: @carolstran You don't need a guard here and below since these sections should be always present

File: src/content/faq/General.md
Comment (IvanGoncharov) [inline src/content/faq/General.md:10]: Idea: We can convert it to MDX and split out sections into separate MD files.
MDX files can include MD files: https://mdxjs.com/getting-started#documents

File: src/components/FAQLayout/index.tsx
Comment (carolstran) [reply src/components/FAQLayout/index.tsx:22]: This is updated in #985 😊

File: src/content/faq/General.md
Comment (carolstran) [reply src/content/faq/General.md:10]: They are all split into separate MD files in #985 - the design in this PR was a placeholder so people could review the content.

Comment (carolstran) [reply src/content/faq/General.md:10]: Converting to MDX could def be an idea for the future though, I can open an issue about it once these PRs are merged!

**Comment by carolstran** [comment]
@IvanGoncharov I noticed you approved #985 and since this PR was mostly for content review, would it be alright if I merged both later today? Then we can discuss any follow-up issues we want to address or open.

File: src/content/faq/General.md
Comment (IvanGoncharov) [inline src/content/faq/General.md:175]: I think it would be more encouraging if we open an issue ourself and link it here.

Comment (carolstran) [reply src/content/faq/General.md:175]: Good idea! I can do that 🚀

File: src/content/faq/General.md
Comment (IvanGoncharov) [inline src/content/faq/General.md:163]: We can also add that if you want to monitor spec proposal you can see them here: https://github.com/graphql/graphql-spec/pulls
Also you can watch proposal being discussed in different working groups here: https://www.youtube.com/channel/UCERcwLeheOXp_u61jEXxHMA

File: src/content/faq/General.md
Comment (IvanGoncharov) [inline src/content/faq/General.md:157]: Maybe worth noting that `description` supports Markdown.

File: src/content/faq/General.md
Comment (IvanGoncharov) [inline src/content/faq/General.md:111]:
```suggestion
Yes, GraphQL is designed to be scalable and is used by many companies in production under a very high load.
```

File: src/content/faq/General.md
Comment (IvanGoncharov) [inline src/content/faq/General.md:69]:
```suggestion
On a typical GraphQL backend, every field on every type has a focused, single-purpose function for resolving that value. Also, instead of trying to handle data batching on the client, [GraphQL moves that logic to the server](/learn/best-practices/#server-side-batching-caching). As a result, there are some inherent performance benefits. Minimizing over-fetching and making [fewer roundtrips to the server](/learn/queries/#fields) are two of them.
```

Comment (IvanGoncharov) [reply src/content/faq/General.md:69]: I would add that GrapQL requests are more complex and can have longer execution times.
That said typical client sends a single GraphQL to get all necessary data so switching to GraphQL should result in the better overall performance of your product.

**Pull request merged**
